### DISCUSSION TITLE
Invalid dbus types?

### DISCUSSION THREAD

**Discussion by michaelwildvarian** [ubuntu/authd#613] [Q&A] [answered]
Hey, I'm new to dbus, so maybe I'm reading the [spec](https://dbus.freedesktop.org/doc/dbus-specification.html#container-types) wrong... But I think that the type `a{s}s` used in [examplebroker/com.ubuntu.auth.ExampleBroker.xml](https://github.com/ubuntu/authd/blob/ffae569ea2ea5383c9f8ce5ab6f01a588fbd42da/examplebroker/com.ubuntu.auth.ExampleBroker.xml#L20) is illegal and should actually read `a{ss}`. At least the codegen tool [https://github.com/tmds/Tmds.DBus](Tmds.DBus.Tool) stumbles over this.

**Comment by adombeck** [comment, marked as answer]
Thanks @michaelwildvarian, I think you're right, `a{s}s` does seems wrong and should probably be `a{ss}` (an array of {key, value} pairs where both key and value are of type string) instead. I'll open a PR for that.

**Reply by adombeck** [reply]
> I'll open a PR for that.

That's https://github.com/ubuntu/authd/pull/619

**Reply by michaelwildvarian** [reply]
Many thanks, @adombeck!
